src/main/java/dk/bec/polonez/reservationsystem/service/OfferService.java

```diff
@@ -35,22 +38,48 @@ public Offer getById(long id) {
     }
 
     public ResponseOfferDto addOffer(CreateOfferDto offerDto) {
-        User user = new User();
-        user.setId(1L);
-
-        Offer.OfferBuilder builder = Offer.builder();
-        Offer offer = builder
+        User owner = userRepository.getById(offerDto.getOwnerId());
+        Offer offer = Offer.builder()
                 .name(offerDto.getName())
-                .owner(user)
+                .owner(owner)
                 .description(offerDto.getDescription())
                 .build();
 
         Offer savedOffer = offerRepository.save(offer);
-        ResponseOfferDto.ResponseOfferDtoBuilder response = ResponseOfferDto.builder();
-        return response
+
+        return ResponseOfferDto.builder()
                 .name(savedOffer.getName())
                 .ownerId(savedOffer.getOwner().getId())
                 .description(savedOffer.getDescription())
                 .build();
     }
+    public ResponseOfferDto updateOffer(long id, CreateOfferDto offerDto) {
+        User owner = userRepository.getById(offerDto.getOwnerId());
+        Offer offer = offerRepository.getById(id);
+
+        offer.setName(offerDto.getName());
+        offer.setOwner(owner);
+        offer.setDescription(offerDto.getDescription());
+
+        Offer updatedOffer = offerRepository.save(offer);
+
+        return ResponseOfferDto.builder()
```

Comment:
Mozecie rozwazyc, czy nie warto tych mapowan gdzies wyniesc. Wg mnie mozna to zrobic na dwa sposoby:
bardziej standardowo - odzielna klasa OfferMapper
mniej standardowo - dodac metode toDto() w klasie Offer